### PR TITLE
[CBRD-23257] Implement the ignore classes mechanism on server

### PR DESCRIPTION
**Pull request by lucianlazar2211**
http://jira.cubrid.org/browse/CBRD-23257

The `--ignore-class-file` parameter was not implemented on the server-side part of loaddb. I have added a check to see if the current class should be ignored based on the file content sent over from the client.

### CODE REVIEW

**Comment by lucianlazar2211** [comment]
This implementation is not mature. We found that it does not work properly so I am working on a better implementation now. :)

**Comment by lucianlazar2211** [comment]
This is still not mature. :D We are going to implement the following:

- [ ] Add the ignore status to the class registry of the said class.

- [ ] Do not use an error to send back to the client when a classname is ignored; better approach would be to extract the information from the class registry and send that information to the client.

- [ ] Make sure that a class being flagged as ignored on the server does not break any checks regarding its content in the registry.

- [ ] Retrieve the information received from the server by changing the signature of `c_handler`.

- [ ] Skip remaining batches when the class is ignored. (This might be implemented actually but needs revision).

- [ ] Remove the new errors on the server side. (Check the `utils.msg` file as well!)

File: src/loaddb/load_common.cpp [hunk 1]
```diff
@@ -547,6 +547,7 @@ namespace cubload
     class_id clsid = FIRST_CLASS_ID;
     batch_id batch_id = NULL_BATCH_ID;
     std::string batch_buffer;
+    bool class_is_ignored = 0;
```
Comment (ribram) [inline src/loaddb/load_common.cpp]:
```suggestion
    bool class_is_ignored = false;
```

File: src/loaddb/load_server_loader.cpp [hunk 2]
```diff
@@ -118,6 +124,13 @@ namespace cubload
 	return;
       }
 
+    // Check if we have to ignore this class.
+    if (is_class_ignored (class_name, true))
```
Comment (ribram) [inline src/loaddb/load_server_loader.cpp]: is_class_ignored looks like a function that does a check, but it does more. it is more clear if the code is something like this:
```c
if (is_class_ignored (class_name))
  {
    m_session.append_log_msg (LOADDB_MSG_IGNORED_CLASS, class_name);
    class_entry *cls_entry = new class_entry (class_name, m_clsid, true);
    m_session.get_class_registry ().register_ignored_class (cls_entry, m_clsid);
    return;
  }
```

File: src/loaddb/load_session.hpp [hunk 3]
```diff
@@ -166,7 +166,7 @@ namespace cubload
     if (get_args ().verbose)
       {
 	std::string log_msg;
-	error_handler::format_log_msg (msg_id, std::forward<Args> (args)...);
+	log_msg = error_handler::format_log_msg (msg_id, std::forward<Args> (args)...);
```
Comment (valerinistor) [inline src/loaddb/load_session.hpp]:
```suggestion
	std::string log_msg = error_handler::format_log_msg (msg_id, std::forward<Args> (args)...);
```

File: src/loaddb/load_common.cpp [hunk 4]
```diff
@@ -600,6 +600,12 @@ namespace cubload
 	    continue;
 	  }
 
+	if (class_is_ignored)
```
Comment (valerinistor) [inline src/loaddb/load_common.cpp]: I think handle_batch from load_common.cpp:644 also should not be called when class is ignored

Comment (valerinistor) [reply src/loaddb/load_common.cpp]: actually I think in this case batch_buffer will be just empty and batch handler will just exit if the buffer is empty

**Pull request merged**